app/controllers/admin/customers_controller.rb

```diff
@@ -80,5 +86,23 @@ def user_can_create_customer?
     def ams_prefix_whitelist
       [:subscription]
     end
+
+    def tag_rule_mapping
+      TagRule.mapping_for(Enterprise.where(id: managed_enterprise_id))
+    end
+
+    # Fetches tags for all customers of the enterprise and returns a hash indexed by customer_id
+    def customer_tags_by_id
+      customer_tags = ::ActsAsTaggableOn::Tag.
+        joins(:taggings).
+        includes(:taggings).
+        where(taggings: { taggable_id: Customer.of(managed_enterprise_id), context: 'tags' })
```

Comment:
I was trying to do this for the inventory page: https://github.com/openfoodfoundation/openfoodnetwork/issues/5131#issuecomment-608345245
nice!

Reply:
Yeah, this was a bit of a pain. Initially I had a look through the repo for the `ActsAsTaggable` gem and there was lots of open issues with people basically screaming "HOW CAN I EAGER-LOAD TAGS??!?!!", but with no resolutions. :see_no_evil:

I think we have to be a bit careful if we try to re-use this bit of code elsewhere, as I think we do some weird modifications to tagging in some places. I think we're abusing the gem a bit already. Basically I think queries that fetch certain tag-related are different depending on the model. There's polymorphic associations and plenty of "automagic" things going on in the background that is really opaque and difficult to follow :see_no_evil:

Reply:
yeah, it's great it's working here!